core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -506,6 +506,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     executors.foreach { eid =>
       removeExecutor(eid, SlaveLost("Stale executor after cluster manager re-registered."))
     }
+    synchronized { executorsPendingToRemove.clear() }
```

Comment:
can we add some comments? usually this should be empty when we reach here, we do this in cases something goes wrong and we must make sure all the states has been reset.

Reply:
`removeExecutor()` is asynchronous, so clear `executorsPendingToRemove` at the end of `reset()` can not work. We should revert this.